File: src/api/news.ts
import { RequestFN } from "~/core/request-function";
import { decodeNews } from "~/decoders/news";
import { encodeDomain } from "~/encoders/domain";
import { type News, type SessionHandle, TabLocation } from "~/models";

export const news = async (session: SessionHandle): Promise<News> => {
  const request = new RequestFN(session, "PageActualites", {
    _Signature_: { onglet: TabLocation.News },

    donnees: {
      modesAffActus: {
        _T: 26,
        V: encodeDomain([0])
      }
    }
  });

  const response = await request.send();
  return decodeNews(response.data.donnees, session);
};

File: library/src/commonMain/kotlin/api/news.kt
package api

import core.RequestFN
import decoders.decodeNews
import encoders.encodeDomain
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.*
import models.News
import models.SessionInformation
import models.TabLocation

suspend fun news (sessionInformation: SessionInformation): News {
    val request = RequestFN(sessionInformation, "PageActualites", Json.encodeToString(buildJsonObject {
        putJsonObject("_Signature_") {
            put("onglet", TabLocation.News.code)
        }

        putJsonObject("donnees") {
            putJsonObject("modesAffActus") {
                put("_T", 26)
                put("V", encodeDomain(listOf(0)))
            }
        }
    }))

    val response = request.send()
    return decodeNews(Json.parseToJsonElement(response.data).jsonObject["donnees"]!!.jsonObject, sessionInformation)
}